### ISSUE TITLE
RTNETLINK answers: Invalid argument when trying to run

### ISSUE THREAD

**Issue by Biggsis** [haugene/docker-transmission-openvpn#1719]
**Before creating this issue I have:**

- [X] Searched for [similar issues](https://github.com/haugene/docker-transmission-openvpn/issues)
- [X] [Read the documentation](https://haugene.github.io/docker-transmission-openvpn/). Especially the troubleshooting section
- [X] Tried to add as much relevant information to the issue as possible

**Describe the problem**

Hello, i have a freshly installed Linux Mint 20.1 on a PC running the latest docker.
I am trying to run a transmission container. But I get an error message RTNETLINK answers: Invalid argument

However, if I OMMIT the LOCAL_NETWORK parameter, then transmissions starts and works and I can access the UI on the local machine. But of course since the local network is missing, I cant access the UI from any other machine in the LAN.

Any help would be greatly appreciated

**Add your docker run command or docker-compose file**

 
 ```
docker run --rm -it --cap-add=NET_ADMIN  
-p 9091:9091   
-e LOCAL_NETWORK=192.168.166.0/16  
-e OPENVPN_PROVIDER=PIA   
-e OPENVPN_CONFIG=france   
-e OPENVPN_USERNAME=xxxxxxx   
-e OPENVPN_PASSWORD=xxxxx   
-e DISABLE_PORT_UPDATER=true   
haugene/transmission-openvpn

 ```

**Logs**

 ```
Starting container with revision: 9d975aba54f79e4fecc57969e51c780a6f9a97af
Creating TUN device /dev/net/tun
Using OpenVPN provider: PIA
Provider PIA has a custom setup script, executing it
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.OEiIAF
Extract OpenVPN config bundle into PIA directory /etc/openvpn/pia
Starting OpenVPN using config spain.ovpn
Modifying /etc/openvpn/pia/spain.ovpn for best behaviour in this container
Setting OpenVPN credentials...
adding route to local network 192.168.166.0/16 via 172.17.0.1 dev eth0
RTNETLINK answers: Invalid argument

WITHOUT the LOCAL_NETWORK parameter the output is with these errors (Locally working):

STARTING TRANSMISSION
Transmission startup script complete.
2021-02-16 08:39:57 WARNING: OpenVPN was configured to add an IPv6 route. However, no IPv6 has been configured for tun0, therefore the route installation may fail or may not work as expected.
2021-02-16 08:39:57 add_route_ipv6(2000::/3 -> :: metric -1) dev tun0
RTNETLINK answers: Permission denied
2021-02-16 08:39:57 ERROR: Linux route -6 add command failed: external program exited with error status: 2
2021-02-16 08:39:57 Initialization Sequence Completed
^C2021-02-16 08:46:09 event_wait : Interrupted system call (code=4)
2021-02-16 08:46:09 delete_route_ipv6(2000::/3)
RTNETLINK answers: No such process
2021-02-16 08:46:09 ERROR: Linux route -6 del command failed: external program exited with error status: 2
2021-02-16 08:46:09 /sbin/ip addr del dev tun0 10.51.112.80/24
2021-02-16 08:46:09 /etc/openvpn/tunnelDown.sh tun0 1500 1553 10.51.112.80 255.255.255.0 init



 ```

**Host system:**

 ```
Server:
 Containers: 3
  Running: 3
  Paused: 0
  Stopped: 0
 Images: 4
 Server Version: 20.10.3
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.4.0-65-generic
 Operating System: Linux Mint 20.1
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.626GiB
 Name: GB-Brix
 ID: XUYM:L3YH:KTRG:ZXNE:4MSJ:PMRG:7WI6:VN3P:WMVS:64R3:GOW4:5GSS
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

 
 ```

**Comment by UniverseXXX** [comment]
I'm getting similar error after the last night container update.

```
DROPPING DEFAULT ROUTE,
2021-02-16 12:51:59 Initialization Sequence Completed,
RTNETLINK answers: No such process,
tinyproxy is on the PATH, run it,
Setting tinyproxy port to 8888,
STARTING TINYPROXY,
Found config file /etc/tinyproxy/tinyproxy.conf, updating settings.,
WARNING: logging deactivated (can't log to stdout when daemonized),
Tinyproxy startup script complete.,
2021-02-16 12:51:59 WARNING: OpenVPN was configured to add an IPv6 route. However, no IPv6 has been configured for tun0, therefore the route installation may fail or may not work as expected.,
2021-02-16 12:51:59 add_route_ipv6(2000::/3 -> :: metric -1) dev tun0,
RTNETLINK answers: Permission denied,
2021-02-16 12:51:59 ERROR: Linux route -6 add command failed: external program exited with error status: 2
2021-02-16 12:51:59 Initialization Sequence Completed
```

**Comment by w0d4** [comment]
Hi, I have exactly the same problem since last nights update.
I'm using bridge as network parameter.
On my container also LOCAL_NETWORK does not make any difference.

**Comment by sorcierwax** [comment]
Same problem here on 2 different servers with PIA too. (Synology Nas and Raspberry PI).
When I put DROP_DEFAULT_ROUTE to False it works though but I guess it's not very secure.

**Comment by pkishino** [comment]
check your routing in the container with ip r.. I know I used to have DROP_DEFAULT_ROUTE true but my vpn provider has updated their servers just a day or two ago and this should be false for me as the route doesn't exist any longer

**Comment by koconnor14** [comment]
Same problem here as well after the container updated yesterday. The transmission daemon does not start and the docker keeps restarting. I ended up using the dev branch in the meantime while we wait on the fix.

**Comment by lightrush** [comment]
Same here. Now does that mean that there is *no route to delete* and therefore removing DROP_DEFAULT_ROUTE is safe or not?

**Comment by pkishino** [comment]
you can check this running
```
ip r
```
in the container and verify the routes
Also, you can try the dnsleaktest script that was linked in some issue
https://github.com/macvk/dnsleaktest

**Comment by pkishino** [comment]
run this inside the container

**Comment by Biggsis** [comment]
You can not run this inside the container, when the container quits with an error message.

**Comment by Biggsis** [comment]
-> pkishino:  I am sorry, I can not run something INSIDE the container if the container does not start at all. If I try to start with the LOCAL_NETWORK parameter, the start aborts with the error message as mentioned above and the container DOES NOT START...
Only without the LOCAL_NETWORK the container starts and of course then the routing is screwed...

**Comment by pkishino** [comment]
your LOCAL_NETWORK looks wrong 192.168.166.0/16 is not valid..this should be 192.168.166.0/24 if you want to limit to last octet

**Comment by pkishino** [comment]
plus you can also trigger the command to run in the container while it is starting up

**Comment by pkishino** [comment]
my bad, I found a bug ..with last commit we already remove the default route so I will remove this statement..

**Comment by pkishino** [comment]
please try with latest

**Comment by cookieisland** [comment]
With the update it started working again (it was broken yesterday for me)

**Comment by koconnor14** [comment]
I am still seeing RTNETLINK answers: No such process in my log. this is the same as what UniverseXXX was seeing / mentioned. I have recreated the container with the LOCAL_NETWORK removed but not change.

The HEALTH check reports that the the network is up but the transmission daemon process is not running. My docker reports an unhealthy state after multiple attempts and then fails. This is restarting at this point. 

Last output | PING google.com (216.58.213.110): 56 data bytes 64 bytes from 216.58.213.110: seq=0 ttl=120 time=289.220 ms 64 bytes from 216.58.213.110: seq=1 ttl=120 time=289.767 ms --- google.com ping statistics --- 2 packets transmitted, 2 packets received, 0% packet loss round-trip min/avg/max = 289.220/289.493/289.767 ms Network is up transmission-daemon process not running
-- | --


If I use the dev tag and recreate the docker this is then stable and the transmission daemon starts up. 

Prior to this update, this has been working and been stable. No DNS leaks when I checked in the past.
